### PR TITLE
Add complier

### PR DESCRIPTION
**Pull request by himanshu1603**
- [x] I followed the contributing guidelines: https://github.com/github/explore/blob/master/CONTRIBUTING.md

I am:
  - [ ] Suggesting edits to an existing topic or collection
  - [x] Curating a new topic or collection

***********EDITING AN EXISTING TOPIC OR COLLECTION************

I'm suggesting these edits to an existing topic or collection:
- [ ] Image (and my file is `*.png`, square, dimensions 288x288)
- [ ] Content (and my changes are in `index.md`)

Please explain why these changes are necessary:

************CURATING A NEW TOPIC************

- [x] I've formatted my changes as a new folder directory, named for the topic as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]`)
- [x] My folder contains a `*.png` image (if applicable) and `index.md`
- [x] All required fields in my `index.md` conform to the Style Guide and API docs: https://github.com/github/explore/tree/master/docs

Please explain why you think this topic should be curated:

************CURATING A NEW COLLECTION************

- [ ] I've formatted my changes as a new folder directory, named for the collection as it appears in the URL on GitHub (e.g. `https://github.com/collections/[NAME]`)
- [ ] My folder contains a `*.png` image (if applicable) and `index.md`
- [ ] All required fields in my `index.md` conform to the Style Guide and API docs: https://github.com/github/explore/tree/master/docs

Please explain why you think this collection should be curated:

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: topics/compiler/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+display_name: Compiler
+short_description: Compiler is a program that translates high level language (human readable) to low level language (machine language).
+topic: compiler
+related: interpreter, gcc, fortran
+wikipedia_url: https://en.wikipedia.org/wiki/Compiler
+---
+A compiler is a software that translate high level language to executable machine code in one go, and reports error if any.The processor executes object code, which indicates when binary high and low signals are required in the arithmetic logic unit of the processor. Examples of compilers include BASIC compilers, fortran complier, C++, C, java compiler, etc.
```
Comment (MikeMcQuaid) [inline topics/compiler/index.md]: - `reports errors (if any)`
- `any.The` needs a space
- Capitalise `fortran` and `java`
- `compilers include`...  you can remove the word `compiler` after that

File: topics/compiler/index.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+display_name: Compiler
+short_description: Compiler is a program that translates high level language (human readable) to low level language (machine language).
```
Comment (MikeMcQuaid) [inline topics/compiler/index.md]: `A compiler is` or `Compilers are`

**Comment by styline** [comment]
Essential the complier

**Comment by styline** [comment]
I mean you build a code translator or I make a mistake?

**Comment by MikeMcQuaid** [comment]
Thanks again @himanshu1603!

**Pull request merged**